pkg/kv/kvserver/rangefeed/catchup_scan.go

```diff
@@ -139,6 +138,51 @@ func (i *CatchUpIterator) CatchUpScan(outputFn outputEventFn, withDiff bool) err
 			break
 		}
 
+		hasPoint, hasRange := i.HasPointAndRange()
+
+		// Emit any new MVCC range tombstones when their start key is encountered.
+		// Range keys can currently only be MVCC range tombstones.
+		//
+		// TODO(erikgrinaker): Find a faster/better way to detect range key changes
+		// that doesn't involve constant comparisons. Pebble probably already knows,
+		// we just need a way to ask it.
+		if hasRange {
+			if rangeBounds := i.RangeBounds(); !rangeBounds.Key.Equal(rangeKeysStart) {
+				// Record the range keys for later processing.
+				rangeKeysStart = append(rangeKeysStart[:0], rangeBounds.Key...)
+				rangeKeys = rangeKeys[:0]
+				for _, rkv := range i.RangeKeys() {
+					rangeKeys = append(rangeKeys, rkv.Clone())
```

Comment:
is there a possibility of this array being huge (memory blowup?)

Reply:
There's an implicit assumption that the number of overlapping range keys will be small. Pebble pulls them all into memory at a given key position, as does the higher layers.